test/runtests.jl

```diff
@@ -68,6 +68,173 @@ end
+    Test.@test coeftable(LR_1c).cols[4][1].v ≈ 0.08982305438748991
+    Test.@test isnan(coeftable(LR_1c).cols[4][2].v)
+    Test.@test coeftable(LR_1c).cols[4][3].v ≈ 0.0
+    Test.@test coeftable(LR_1c).cols[4][4].v ≈ 0.12938683057308423
+
+    OLS_2 = fit(
+        LinearModel,
+        @formula(label_2 ~ 1 + feature_1 + feature_2 + feature_3),
+        df,
+        true)
+    Test.@test coef(OLS_2)[1] ≈ -0.18433687795387846
+    Test.@test coef(OLS_2)[2] ≈ 0.0
+    Test.@test coef(OLS_2)[3] ≈ 6.937516672427037e-6
+    Test.@test coef(OLS_2)[4] ≈ 0.0002690550151709414
+    Test.@test stderr(OLS_2)[1] ≈ 0.008314840332069603
+    Test.@test isnan(stderr(OLS_2)[2])
+    Test.@test stderr(OLS_2)[3] ≈ 5.390720287418552e-6
+    Test.@test stderr(OLS_2)[4] ≈ 2.1633280582963663e-6
+    Test.@test coeftable(OLS_2).cols[4][1].v ≈ 2.399616832216088e-106
+    Test.@test isnan(coeftable(OLS_2).cols[4][2].v)
+    Test.@test coeftable(OLS_2).cols[4][3].v ≈ 0.19814609164909394
+    Test.@test coeftable(OLS_2).cols[4][4].v ≈ 0.0
+
+    LR_2a = fit(
+        GeneralizedLinearModel,
+        @formula(label_2 ~ 1 + feature_1  + feature_3),
+        df,
+        GLM.Binomial(),
+        GLM.LogitLink())
+    Test.@test coef(LR_2a)[1] ≈ -5.794539719571302
+    Test.@test coef(LR_2a)[2] ≈ 0.00013965128998063396
+    Test.@test coef(LR_2a)[3] ≈ 0.0022644689769415224
+    Test.@test stderr(LR_2a)[1] ≈ 0.1319550070864715
+    Test.@test stderr(LR_2a)[2] ≈ 0.0001167815891984272
+    Test.@test stderr(LR_2a)[3] ≈ 4.438988363503985e-5
+    Test.@test coeftable(LR_2a).cols[4][1].v ≈ 0.0
+    Test.@test coeftable(LR_2a).cols[4][2].v ≈ 0.23176169574597333
+    Test.@test coeftable(LR_2a).cols[4][3].v ≈ 0.0
+
+    LR_2b = fit(
+        GeneralizedLinearModel,
+        @formula(label_2 ~ 1  + feature_2 + feature_3),
+        df,
+        GLM.Binomial(),
+        GLM.LogitLink())
+    Test.@test coef(LR_2b)[1] ≈ -5.794539719571302
+    Test.@test coef(LR_2b)[2] ≈ 6.982564499031698e-5
+    Test.@test coef(LR_2b)[3] ≈ 0.0022644689769415224
+    Test.@test stderr(LR_2b)[1] ≈ 0.1319550070864715
+    Test.@test stderr(LR_2b)[2] ≈ 5.83907945992136e-5
+    Test.@test stderr(LR_2b)[3] ≈ 4.438988363503985e-5
+    Test.@test coeftable(LR_2b).cols[4][1].v ≈ 0.0
+    Test.@test coeftable(LR_2b).cols[4][2].v ≈ 0.23176169574597333
+    Test.@test coeftable(LR_2b).cols[4][3].v ≈ 0.0
+
+    Test.@test_throws(
+        LinearAlgebra.PosDefException,
+        fit(
+            GeneralizedLinearModel,
+            @formula(label_2 ~ 1 + feature_1 + feature_2 + feature_3),
+            df,
+            Binomial(),
+            GLM.LogitLink())
+        )
+
+    LR_2c = fit(
+        GeneralizedLinearModel,
+        @formula(label_2 ~ 1 + feature_1 + feature_2 + feature_3),
+        df,
+        GLM.Binomial(),
+        GLM.LogitLink(),
+        allowrankdeficient=true)
+
+    # Julia 0.7 and Julia 1.0 give different values for the coefficients
+    # and standard errors.
```

Comment:
Comment on why if possible.

Reply:
Oh god I wish I knew why. I tried experimenting a little bit to see if I could figure out why this happens, but I think that I don't have the requisite knowledge to understanding what's going on.

Reply:
Maybe something changed in the underlying libraries? Anyway, better do `if VERSION >= 1.0` and use two distinct blocks. But are these two solutions correct?

BTW, why not check the deviance for each model?